src/tools/cephfs/top/cephfs-top

```diff
@@ -205,25 +206,35 @@ class FSTop(object):
         for item in MAIN_WINDOW_TOP_LINE_ITEMS_END:
             coord = x_coord_map[item]
             if item == FS_TOP_MAIN_WINDOW_COL_MNTPT_HOST_ADDR:
-                self.mainw.addstr(y_coord, coord[0],
-                                  f'{client_meta[CLIENT_METADATA_MOUNT_POINT_KEY]}@'
-                                  f'{client_meta[CLIENT_METADATA_HOSTNAME_KEY]}/'
-                                  f'{client_meta[CLIENT_METADATA_IP_KEY]}')
+                if CLIENT_METADATA_MOUNT_POINT_KEY in client_meta and \
+                   CLIENT_METADATA_HOSTNAME_KEY in client_meta and \
+                   CLIENT_METADATA_IP_KEY in client_meta:
+                    self.mainw.addstr(y_coord, coord[0],
+                                      f'{client_meta[CLIENT_METADATA_MOUNT_POINT_KEY]}@'
+                                      f'{client_meta[CLIENT_METADATA_HOSTNAME_KEY]}/'
+                                      f'{client_meta[CLIENT_METADATA_IP_KEY]}')
         for item in MAIN_WINDOW_TOP_LINE_ITEMS_START:
+            if CLIENT_METADATA_VALIDMETRICS_KEY not in client_meta:
+                # eliminate invalid CLIENT_ID display
+                continue
             coord = x_coord_map[item]
             hlen = coord[1] - len(ITEMS_PAD)
             if item == FS_TOP_MAIN_WINDOW_COL_CLIENT_ID:
                 self.mainw.addstr(y_coord, coord[0],
                                   wrap(client_id.split('.')[1], hlen))
-            elif item == FS_TOP_MAIN_WINDOW_COL_MNT_ROOT:
+            elif (item == FS_TOP_MAIN_WINDOW_COL_MNT_ROOT and
```

Comment:
nit: can reuse the helper routine mentioned above. something like:

```
def has_metrics(client_meta, metrics_key):
    """Check if client_meta has metrics_key"""
```

Then for `FS_TOP_MAIN_WINDOW_COL_MNTPT_HOST_ADDR` you could pass a composite value that checks existence of mount_point, hostname and IP keys.